pysr/sr.py

```diff
@@ -2586,3 +2583,190 @@ def _mutate_parameter(param_name: str, param_value):
         return False
 
     return param_value
+
+
+class PySRSequenceRegressor(PySRRegressor):
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        self.recursive_history_length = recursive_history_length
+        super().__init__(**kwargs)
+
+    def fit(
+        self,
+        X,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+    ) -> "PySRRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_samples, 1) or (1, n_samples).
+        Xresampled : ndarray | pandas.DataFrame
+            Resampled training data, of shape (n_resampled, 1) or (1, n_resampled),
+            to generate a denoised data on. This
+            will be used as the training data, rather than `X`.
+        weights : ndarray | pandas.DataFrame
+            Weight array of the same shape as `X`, but not for the
+            first recurrence_history_length terms. Therefore, the shape is
+            (n_samples-recurrence_history_length, 1) or (1, n_samples-recurrence_history_length)
+            Each element is how to weight the mean-square-error loss
+            for that particular element of `X`. Alternatively,
+            if a custom `loss` was set, it will can be used
+            in arbitrary ways.
+        variable_names : list[str]
+            A list of names for the variables, rather than "x0", "x1", etc.
+            If `X` is a pandas dataframe, the column names will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to recurrence_history_length.
```

Comment:
I think these should be automatically generated and include information about relative time. For example, `X_{t-1}`, `X_{t-2}` might be good choices?